docs/proposals/env-vars.md

```diff
@@ -0,0 +1,316 @@
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [docs](/docs/)
+
+## Open Questions [optional]
+
+> 1. API change
+
+Assuming this gets delivered while Shipwright builds is still v1alpha1, we just have to announce to users
+what new API is provided.
+
+But if the implementation of this is delayed to the point that we are at say v1beta1, then we have to 
+consider the round trip experience when a client is at a different version of the server as discussed in
+[Kubernetes Deprecation Policy Rule #2](https://kubernetes.io/docs/reference/using-api/deprecation-policy/).
+And yes, adding an API is not the same as deprecating one, but the round trip guarantee applies (as Tekton
+learnt recently).  The use of an annotation that captures version differences comes into play.
+
+> 2. Order of precedence
+
+Similar to other shipwright enhancements currently in flight, the notion of settings at a `BuildStrategy` level
+serving as a default, with the ability to override those defaults at a more granular `Build` or `BuildRun`, resonates
+with our topic here of managing environment various in Shipwright.  This order makes sense with developer flows for
+example.
+
+That said, one could also see cluster administrators wanting certain environment variables to never be set in their
+cluster, don't use certain values, or that certain environment variables always use certain values.  
+Perhaps certain values for an environment variable, or the existence of the environment variable at all,
+enable features in underlying tools encapsulated by a given `BuildStrategy` that are deemed inappropriate, unsafe,
+of too costly resource wise for the cluster in question.
+
+There are a few possibilities in how the API is or is not structured that could allow for support of each of the above
+concerns.  We should get to a consensus on which way(s) we support in the enhancement proposal. Ideally, one approach
+makes for a simpler implementation path.  If multiple approaches seem necessary, then we have to sort out if those 
+varying approaches can be staged, or have to be implemented all at once.
+
+## Summary
+
+One of the fields in the core Kubernetes API, Pods and their Containers, are Environment Variables.  Of course those
+translate to operating system level environment variables being set when running the same commands executed
+in Pod Containers from the command line of their personal computuer.
+
+The next major dependency for Shipwright, Tekton, has a vast enough array of scenarios it addresses that it provides a 
+direct line to the Kubernetes Container API in their API.  As as result, environment variable fields from `Steps`/`Containers`
+are directly accessible from Tekton API.
+
+Specifically [Tekton TaskRuns](https://github.com/tektoncd/pipeline/blob/main/docs/taskruns.md) reference
```

Comment:
Steps are defined in the spec of a Task, not a TaskRun.

Reply:
yes but there is a ptr to a TaskSpec  in the TaskRun, and the Steps are in the TaskSpec definition.  

if that is not clear my text above I'll add more verbiage

Reply:
ok finally back to this today .... even if you and I are okay,  I'm making the relationship chain more explicit and clear in my next update